src/battle/partner/parakarry.c

```diff
@@ -1,7 +1,336 @@
 #include "common.h"
 #include "battle/battle.h"
+#include "hud_element.h"
+#include "effects.h"
 
-INCLUDE_ASM(s32, "battle/partner/parakarry", func_80238000_6FFD80);
+extern s32 D_8023BD78[7];
+extern s32 D_8023BD98[6];
+extern s32 D_8023BDB4;
+extern s32 D_8023BDB8;
+extern s32 D_8023BDBC;
+extern s32 D_8023BDC0;
+extern s32 D_8023BDC4;
+extern s32 D_8023BDC8;
+extern HudScript HudScript_AimMarkerA[0];
+extern HudScript HudScript_AimMarkerB[0];
+extern HudScript HudScript_AimMarkerC[0];
+extern HudScript HudScript_AimMarkerD[0];
+extern HudScript HudScript_AimMarkerE[0];
+extern HudScript HudScript_AimMarkerF[0];
+extern HudScript HudScript_AimReticle[0];
+extern HudScript HudScript_AimShimmerA[0];
+extern HudScript HudScript_AimShimmerB[0];
+extern HudScript HudScript_AimShimmerC[0];
+extern HudScript HudScript_AimShimmerD[0];
+extern HudScript HudScript_AimShimmerE[0];
+extern HudScript HudScript_AimShimmerF[0];
+extern HudScript HudScript_AimTarget[0];
+extern HudScript HudScript_StickHoldLeft[0];
+extern HudScript HudScript_StickNeutral[0];
+
+extern f64 D_8023BD60_703AE0;
+
+ApiStatus func_80238000_6FFD80(Evt* evt, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    Actor* parakarry = battleStatus->partnerActor;
+    ActorState* state = &parakarry->state;
+    Actor* targetActor;
+    ActorPart* targetActorPart;
+    f32 x, y, z;
+    s32 screenX, screenY, screenZ;
+    f32 tempX, tempY, tempZ;
+    f32 clampedAngleDiff, tempAngle;
+    s32 hudID, i;
+
+    if(isInitialCall) {
+        evt->functionTemp[0] = 0;
+    }
+
+    switch(evt->functionTemp[0]) {
+        case 0:
+            hudID = hud_element_create(HudScript_AimMarkerF);
+            D_8023BD78[0] = hudID;
+            hud_element_set_render_depth(hudID, 0xA);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimMarkerE);
+            D_8023BD78[1] = hudID;
```

Comment:
Might be possible to compress this to
```suggestion
            D_8023BD78[1] = hudID = hud_element_create(HudScript_AimMarkerE);
```